src/cljam/io/util/bin.clj

```diff
@@ -37,8 +40,18 @@
 (defn get-spans
   "Calculate span information for random access from index data such as tabix."
   [index-data ^long ref-idx ^long beg ^long end]
-  (let [bins (reg->bins beg end)
-        chunks (get-chunks index-data ref-idx bins)
-        min-offset (get-min-offset index-data ref-idx beg)]
-    (->> (util-chunk/optimize-chunks chunks min-offset)
-         (map vals))))
+  (let [bins
+        (reg->bins beg end (get-min-shift index-data) (get-depth index-data))
+        chunks-list (get-chunks index-data ref-idx bins)
+        min-offsets (get-min-offset index-data ref-idx beg bins)]
+    (->>
+     (mapcat
+      (fn [chunks min-offset]
+        (filter
+         (fn [^Chunk x]
+           (> (.end x) min-offset))
+         chunks))
+      chunks-list
+      min-offsets)
```

Comment:
I don't get the point of comparing min-offset to chunks in the same bin.
I think a min-offset of the smallest bin with the smallest index is sufficient for all chunks.

```
|=======================0=======================|
|-----------1-----------|===========2===========|
|-----3-----|-----4-----|=====5=====|=====6=====|
|--7--|--8--|--9--|--10-|--11-|==12=|==13=|==14=|
                           |-x-|
                    |---y---|
                    |-----y'-----|

                                   |====q====|
```

Let's consider a binary tree instead of an octree.
Given a query interval `q`, chunks in overlapping bins `[0, 2, 5, 6, 12, 13, 14]` will be candidates.
Let `x` (which itself belongs to the bin `5`) be an entity with the smallest starting position overlapping the bin `12`. 
Chunks ending before the starting file offset of `x` in the bins `[0, 2, 5, 6]` can be ignored since they never overlap the bin `12`. (e.g. `y` in the bin `0` can be ignored)
Entities like `y'` never exist otherwise the l-offset of the bin `12` would be `y'` instead of `x`.

We can apply this method recursively for each depth (use l-offset of `12` for filtering chunks in the bins `[0, 2, 5, 6]`, `5` for `[0, 2]`...), but I think it's unnecessary for now.